From 142929332777fca08b587920438c42523406d3ed Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:48:07 -0400
Subject: [iommu] vt-d: Add ACPI namespace device reporting structures

Message-id: <20140905014807.31297.53345.stgit@gir.stowe>
Patchwork-id: 91316
O-Subject: [RHEL7.1 PATCH 23/65] iommu/vt-d: Add ACPI namespace device reporting structures
Bugzilla: 1129880 1087643
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1087643
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129880
Upstream Status: 86a54dcce6ccb0d4a5810ed8112011039d855bee

commit 86a54dcce6ccb0d4a5810ed8112011039d855bee
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Fri Mar 7 12:43:40 2014 +0000

    iommu/vt-d: Add ACPI namespace device reporting structures

    Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/acpi/actbl2.h b/include/acpi/actbl2.h
index 40f7ed1..07840ed 100644
--- a/include/acpi/actbl2.h
+++ b/include/acpi/actbl2.h
@@ -419,7 +419,8 @@ enum acpi_dmar_type {
  ACPI_DMAR_TYPE_RESERVED_MEMORY = 1,
  ACPI_DMAR_TYPE_ATSR = 2,
  ACPI_DMAR_HARDWARE_AFFINITY = 3,
- ACPI_DMAR_TYPE_RESERVED = 4 /* 4 and greater are reserved */
+ ACPI_DMAR_TYPE_ANDD = 4,
+ ACPI_DMAR_TYPE_RESERVED = 5 /* 5 and greater are reserved */
 };
 
 /* DMAR Device Scope structure */
@@ -440,7 +441,8 @@ enum acpi_dmar_scope_type {
  ACPI_DMAR_SCOPE_TYPE_BRIDGE = 2,
  ACPI_DMAR_SCOPE_TYPE_IOAPIC = 3,
  ACPI_DMAR_SCOPE_TYPE_HPET = 4,
- ACPI_DMAR_SCOPE_TYPE_RESERVED = 5 /* 5 and greater are reserved */
+ ACPI_DMAR_SCOPE_TYPE_ACPI = 5,
+ ACPI_DMAR_SCOPE_TYPE_RESERVED = 6 /* 6 and greater are reserved */
 };
 
 struct acpi_dmar_pci_path {
@@ -502,6 +504,15 @@ struct acpi_dmar_rhsa {
  u32 proximity_domain;
 };
 
+/* 4: ACPI Namespace Device Declaration Structure */
+
+struct acpi_dmar_andd {
+ struct acpi_dmar_header header;
+ u8 reserved[3];
+ u8 device_number;
+ u8 object_name[];
+};
+
 /*******************************************************************************
  *
  * HPET - High Precision Event Timer table
-- 
1.7.1